pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/BaseDistinctAggregateAggregationFunction.java

```diff
@@ -147,7 +149,7 @@ protected void svAggregate(int length, AggregationResultHolder aggregationResult
 
     // For dictionary-encoded expression, store dictionary ids into the bitmap
     Dictionary dictionary = blockValSet.getDictionary();
-    if (dictionary != null) {
+    if (dictionary != null && !_nullHandlingEnabled) {
```

Comment:
We can still use dictionary id even when null handling is enabled

Reply:
We cannot use dictionary ID when NULL handling is enabled because Integer.MIN_VALUE and NULL share the same dictionary ID.

Reply:
That is not true here. We won't add null values into the set, so the one added to the set is guaranteed to be `Integer.MIN_VALUE`

Reply:
Done.

I previously thought the `int[] dictIds = blockValSet.getDictionaryIdsSV();` was to get the dictionary IDs metadata and had the confusion in the earlier comment.